openedx/openedx-platform
# [SE-4473] Add completion progress bar to dashboard

## Description

This PR adds the required changes to the edx-platform to enable displaying completion progress bar. The actual progress bar is implemented in [openedx-completion-aggregator](https://github.com/open-craft/openedx-completion-aggregator) django app. The changes here only display it in an iframe on the dashboard. It is also put behind a feature flag, `SHOW_PROGRESS_BAR` to prevent any changes to the default behaviour.

## Screenshot:

![progess-bars-examples-final](https://user-images.githubusercontent.com/13742492/140732725-8755156c-7288-4323-abec-68b1384ae932.png)


## Supporting information

JIRA: [SE-4473](https://tasks.opencraft.com/browse/SE-4473)
Sandbox URL: TBD
Required configuration: https://github.com/edx/configuration/pull/6609


## Testing instructions

1. Setup your devstack with this branch.
2. Set the feature flag `SHOW_PROGRESS_BAR` to true under `FEATURES` in lms.yml and studio.yml.
3. Add the `COMPLETION_AGGREGATOR_URL` setting in lms.yml as shown below:
```
COMPLETION_AGGREGATOR_URL: /completion-aggregator/progress_bar
```
or ensure the configuration repo used for deployment includes https://github.com/edx/configuration/pull/6609.
4. Ensure completion tracking is enabled.
5. Pip install completion-aggregator django app containing https://github.com/open-craft/openedx-completion-aggregator/pull/89.
6. Start new course in LMS and verify that both progress bars are visible and updates correctly.

## Reviewers

- [ ] edX reviewer: TDB
